src/XR/features/WebXRNearInteraction.ts

```diff
@@ -0,0 +1,590 @@
+        return mesh.isEnabled() && mesh.isVisible && mesh.isPickable && mesh.isNearPickable;
+    }
+
+    /**
+     * Filter used for near interaction garb
+     */
+    private nearGrabPredicate(mesh: AbstractMesh): boolean {
+        return mesh.isEnabled() && mesh.isVisible && mesh.isPickable && mesh.isNearGrabbable;
+    }
+
+    private readonly _hoverRadius = 0.1;
+    private readonly _pickRadius = 0.03;
+    private _indexTipQuaternion = new Quaternion();
+    private _indexTipOrientationVector = Vector3.Zero();
+
+    protected _onXRFrame(_xrFrame: XRFrame) {
+        Object.keys(this._controllers).forEach((id) => {
+            // only do this for the selected pointer
+            const controllerData = this._controllers[id];
+            if (!this._options.enablePointerSelectionOnAllControllers && id !== this._attachedController) {
+                controllerData.pick = null;
+                return;
+            }
+            if (!controllerData.nearGrabInProcess) {
+                controllerData.nearInteraction = false;
+                controllerData.nearGrab = false;
+            }
+            // Every frame check collisions/input
+            if (controllerData.xrController) {
+                const hand = controllerData.xrController.inputSource.hand;
+                if (hand) {
+                    const xrIndexTip = hand.get("index-finger-tip");
+                    if (xrIndexTip) {
+                        let indexTipPose = _xrFrame.getJointPose!(xrIndexTip, this._xrSessionManager.referenceSpace);
+                        if (indexTipPose && indexTipPose.transform) {
+                            let zAxisRHSMultiplier = this._scene.useRightHandedSystem ? 1 : -1;
+                            const indexTipPos = indexTipPose.transform.position;
+                            const indexTipOrientation = indexTipPose.transform.orientation;
+                            this._indexTipQuaternion.set(
+                                indexTipOrientation.x,
+                                indexTipOrientation.y,
+                                indexTipOrientation.z * zAxisRHSMultiplier,
+                                indexTipOrientation.w * zAxisRHSMultiplier
+                            );
+
+                            // set positions for near pick and hover
+                            if (controllerData.pickIndexMeshTip) {
+                                controllerData.pickIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, indexTipPos.z * zAxisRHSMultiplier);
+                            }
+                            if (controllerData.hoverIndexMeshTip) {
+                                controllerData.hoverIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, indexTipPos.z * zAxisRHSMultiplier);
+                            }
+
+                            // set near interaction grab ray parameters
+                            const nearGrabRayLength = 5 * this._hoverRadius;

```

Comment:
>5 [](http://example.com/codeflow?start=54&length=1)

nit: make this (5) or nearGrabRayLength a const value up alongside this._hoverRadius